test/server/util/csv_helper.js

```diff
@@ -0,0 +1,116 @@
+import * as chai from 'chai';
+import fs from 'fs';
+import path from 'path';
+import { validateParseCSV, extraHeadersError as extraHeaders, missingHeadersError as missingHeaders } from '../../../server/util/csv_valid_parse';
+
+const checkObjectProps = (expectedProps, obj) => {
+  return expectedProps.reduce((accum, curr) => {
+    const propertyExists = Object.prototype.hasOwnProperty.call(obj, curr);
+    return accum && propertyExists;
+  }, true);
+};
+
+const paths = {
+  extraHeaders: 'extra_headers_contact_list.csv',
+  missingHeaders: 'missing_header_contact_list.csv',
+  completeHeaders: 'contact_list.csv'
+};
+
+const { extraHeaders: extra, missingHeaders: missing, completeHeaders: complete } = paths;
+
+const mockBufferExtraHeaders = fs.readFileSync(path.resolve(__dirname, extra));
+const mockBufferMissingHeader = fs.readFileSync(path.resolve(__dirname, missing));
+const mockBuffer = fs.readFileSync(path.resolve(__dirname, complete));
+
+const { expect } = chai;
+
+describe('CSV validation and parsing helper function tests: ', () => {
+  console.log('now running CSV validation tests: ');
+  describe('validation error handling: ', () => {
+    it('when passed a csv with headers missing, it should throw the "missingHeaders error" ', (done) => {
+      validateParseCSV({ data: mockBufferMissingHeader })
+        .then((contactObjs) => {
+          expect(contactObjs).to.equal(undefined);
+        })
+        .catch((err) => {
+          expect(JSON.stringify(err)).to.equal(JSON.stringify(missingHeaders(['email'])));
+        });
+      done();
```

Comment:
This done() should be within the catch block since we're expecting an error.
Alternatively, we can even use assert.throws (http://chaijs.com/api/assert/#method_throws)
Example of how to use it: https://stackoverflow.com/questions/14966821/testing-for-errors-thrown-in-mocha
`
assert.throws(() => validateParseCSV({ data: mockBufferMissingHeader }), missingHeaders(['email'])`